src/aws_iot_mqtt_client_subscribe.c

```diff
@@ -260,8 +260,8 @@ static IoT_Error_t _aws_iot_mqtt_internal_subscribe(AWS_IoT_Client *pClient, con
 	//	return RX_MESSAGE_INVALID_ERROR;
 	//}
 
-	pClient->clientData.messageHandlers[indexOfFreeMessageHandler].topicName =
-			pTopicName;
+	memcpy(pClient->clientData.messageHandlers[indexOfFreeMessageHandler].topicName,
```

Comment:
Need to check that `topicNameLen < TOPIC_NAME_MAX_LENGTH`.  Otherwise, this introduces a memory corruption bug.